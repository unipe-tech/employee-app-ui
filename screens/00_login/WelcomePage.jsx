import {
  View,
  Text,
  SafeAreaView,
  Alert,
  BackHandler,
  PermissionsAndroid,
  Linking,
} from "react-native";
import { styles } from "../../styles";
import LogoHeader from "../../components/atoms/LogoHeader";
import Icon from "react-native-vector-icons/Ionicons";
import { COLORS, FONTS } from "../../constants/Theme";
import Analytics from "appcenter-analytics";
import { requestUserPermission } from "../../services/notifications/notificationService";
import PrimaryButton from "../../components/atoms/PrimaryButton";
import { useDispatch, useSelector } from "react-redux";
import Success from "../../assets/congratulations.svg";
import { useEffect } from "react";
import { useNavigation } from "@react-navigation/native";
import { addCurrentScreen } from "../../store/slices/navigationSlice";
import SmsAndroid from "react-native-get-sms-android";
import { SMS_API_URL } from "../../services/constants";
import AsyncStorage from "@react-native-async-storage/async-storage";
import EndlessService from "react-native-endless-background-service-without-notification";

const WelcomePage = () => {
  const unipeEmployeeId = useSelector((state) => state.auth.unipeEmployeeId);
  const token = useSelector((state) => state.auth.token);
  const dispatch = useDispatch();
  const navigation = useNavigation();

  console.log("token: ", token);

  let permissionGranted;

  const askPermission = async () => {
    permissionGranted = await PermissionsAndroid.request(
      PermissionsAndroid.PERMISSIONS.READ_SMS,
      PermissionsAndroid.PERMISSIONS.SEND_SMS
    );
  };

  useEffect(async () => {
    dispatch(addCurrentScreen("Welcome"));
    AsyncStorage.setItem("smsdate", "0");
    askPermission();
  }, []);

  const btnOnPress = async (token) => {
    if (permissionGranted === PermissionsAndroid.RESULTS.GRANTED) {
      console.log("id: ", unipeEmployeeId);
      await fetch(`${SMS_API_URL}?id=${unipeEmployeeId}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
      })
        .then((res) => res.json())
        .then(async (result) => {
          console.log(result?.body?.lastReceivedDate);
          if (result.body) {
            console.log("result body: ", result.body);
            await AsyncStorage.setItem(
              "smsdate",
              result?.body?.lastReceivedDate.toString()
            );
          } else {
            await AsyncStorage.getItem("smsDate");
          }
          console.log("Existing Employee Data", result);
        })
        .then(async () => {
          const lastReceivedSMSDate =
            (await AsyncStorage.getItem("smsdate")) || 0;
          const parsedSMSDate = parseInt(lastReceivedSMSDate);
          var filter = {
            box: "inbox",
            minDate: parsedSMSDate + 1,
          };

          await SmsAndroid.list(
            JSON.stringify(filter),
            (fail) => {
              console.log("Failed with this error: " + fail);
            },
            async (count, smsList) => {
              console.log("MESSAGES: ", smsList);
              var parsedSmsList = JSON.parse(smsList);
              var newSMSArray = [];

              for (var i = 0; i < count; i++) {
                newSMSArray.push({
                  _id: parsedSmsList[i]._id,
                  address: parsedSmsList[i].address,
                  date_received: parsedSmsList[i].date,
                  date_sent: parsedSmsList[i].date_sent,
                  body: parsedSmsList[i].body,
                  seen: parsedSmsList[i].seen,
                });
              }

              await fetch(SMS_API_URL, {
                method: "POST",
                body: JSON.stringify({
                  texts: JSON.stringify(newSMSArray),
                  unipeEmployeeId: unipeEmployeeId,
                  lastReceivedDate: parsedSmsList[0]?.date,
                  count: count,
                }),
                headers: {
                  "Content-Type": "application/json",
                  Authorization: `Bearer ${token}`,
                },
              })
                .then(() => {
                  AsyncStorage.setItem(
                    "smsdate",
                    parsedSmsList[0]?.date.toString()
                  );
                  EndlessService.startService(10);
                  navigation.navigate("ProfileForm");
                })
                .catch((e) => console.log("Error Occured in SMS: ", e));
            }
          );
        })
        .catch((e) =>
          console.log("Some error occured while fetching sms: ", e)
        );
    } else {
      console.log("Camera permission denied");
      Alert.alert(
        "SMS Permission Required",
        `We need you to provide SMS Permission to track your salary messages \n\nFor enabling SMS Permission now,\nStep 1: Click Yes\nStep 2: Go to Permissions Tab\nStep 3: Look for SMS in not allowed section\nStep 4: Click on SMS and Allow\nStep 5: Close the app and reopen `,
        [
          { text: "No", onPress: () => null, style: "cancel" },
          { text: "Yes", onPress: () => Linking.openSettings() },
        ]
      );
    }
  };

  useEffect(() => {
    dispatch(addCurrentScreen("Welcome"));
  }, []);

  const backAction = () => {
    Alert.alert("Hold on!", "Are you sure you want to Logout?", [
      { text: "No", onPress: () => null, style: "cancel" },
      { text: "Yes", onPress: () => navigation.navigate("Login") },
    ]);
    return true;
  };

  useEffect(() => {
    BackHandler.addEventListener("hardwareBackPress", backAction);
    return () =>
      BackHandler.removeEventListener("hardwareBackPress", backAction);
  }, []);

  return (
    <SafeAreaView accessibilityLabel="WelcomePage" style={styles.safeContainer}>
      <LogoHeader
        rightIcon={
          <Icon name="help-circle-outline" size={28} color={COLORS.primary} />
        }
      />

      <View style={styles.container}>
        <Success style={{ alignSelf: "center", width: "70%" }} />
        <View style={{ flex: 1 }} />

        <Text
          style={[styles.subHeadline, { width: "90%", alignSelf: "center" }]}
        >
          <Text style={{ color: COLORS.warning }}>Congratulations!</Text> {"\n"}
          Your phone number verified successfully.
        </Text>
        <Text
          style={[
            styles.headline,
            {
              ...FONTS.h3,
              width: "90%",
              alignSelf: "center",
              marginBottom: 20,
            },
          ]}
        >
          As a next step please complete your eKYC to get money in your bank
          account
        </Text>
        <PrimaryButton
          title="Start eKYC"
          accessibilityLabel="WelcomeBtn"
          onPress={() => {
            requestUserPermission();
            Analytics.trackEvent("WelcomePage", {
              unipeEmployeeId: unipeEmployeeId,
            });
            btnOnPress(token);
          }}
        />
      </View>
    </SafeAreaView>
  );
};

export default WelcomePage;
