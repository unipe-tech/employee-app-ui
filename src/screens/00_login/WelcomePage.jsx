import { useNavigation } from "@react-navigation/core";
import Analytics from "appcenter-analytics";
import { Linking, SafeAreaView, Text, View } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import { useDispatch, useSelector } from "react-redux";
import Clock from "../../assets/Clock.svg";
import InterestFree from "../../assets/InterestFree.svg";
import OnDemand from "../../assets/OnDemand.svg";
import LogoHeader from "../../components/atoms/LogoHeader";
import PrimaryButton from "../../components/atoms/PrimaryButton";
import ShieldTitle from "../../components/atoms/ShieldTitle";
import SvgListItem from "../../components/molecules/SvgListItem";
import { COLORS, FONTS } from "../../constants/Theme";
import { useEffect } from "react";
import { requestUserPermission } from "../../services/notifications/notificationService";
import { onboardingStyles, styles } from "../../styles";
import { addCurrentScreen } from "../../store/slices/navigationSlice";

const WelcomePage = () => {
  const dispatch = useDispatch();
  const navigation = useNavigation();
  useEffect(() => {
    dispatch(addCurrentScreen("Welcome"));
  }, []);
  const unipeEmployeeId = useSelector((state) => state.auth.unipeEmployeeId);

  const data = [
    {
      title: "On-demand Salary\n(पाएँ वेतन अपने मनचाहे समय पर)",
      imageUri: <OnDemand />,
    },
    {
      title: "Interest Free\nशून्य ब्याज दर",
      imageUri: <Clock />,
    },
    {
      title:
        "Money in your bank in 5 mins\nसिर्फ़ पाँच मिनट में पैसा आपके बैंक अकाउंट में",
      imageUri: <InterestFree />,
    },
  ];

  return (
    <SafeAreaView style={styles.safeContainer}>
      <LogoHeader
        rightIcon={
          <Icon name="logo-whatsapp" size={28} color={COLORS.primary} />
        }
        rightOnPress={() => {
          Linking.openURL(`whatsapp://send?text=&phone=7483447528`);
        }}
      />
      <View style={styles.container}>
        <Text
          style={{
            ...FONTS.title,
            color: COLORS.primary,
          }}
        >
          नमस्ते
        </Text>
        <View style={styles.row}>
          <View style={onboardingStyles.curvedBox} />
          <Text style={{ ...FONTS.h2, color: COLORS.secondary }}>
            Unipe के साथ पाएँ
          </Text>
        </View>
        {data.map((item, index) => (
          <SvgListItem item={item} key={index} />
        ))}

        <View style={{ flex: 1 }} />
        <ShieldTitle title="100% Secure" />
        <PrimaryButton
          title="Get Started Now"
          onPress={() => {
            requestUserPermission();
            Analytics.trackEvent("WelcomePage", {
              unipeEmployeeId: unipeEmployeeId,
            });
            navigation.navigate("HomeStack");
          }}
        />
      </View>
    </SafeAreaView>
  );
};

export default WelcomePage;
