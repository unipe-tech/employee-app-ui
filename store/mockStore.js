const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
