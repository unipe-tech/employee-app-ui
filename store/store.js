import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";
import aadhaarSlice from "./slices/aadhaarSlice";
import panSlice from "./slices/panSlice";
import bankSlice from "./slices/bankSlice";
import esicSlice from "./slices/esicSlice";
import profileSlice from "./slices/profileSlice";
import navigationSlice from "./slices/navigationSlice";

export const store = configureStore({
  reducer: {
    auth: authSlice,
    aadhaar: aadhaarSlice,
    pan: panSlice,
    bank: bankSlice,
    profile: profileSlice,
    esic: esicSlice,
    navigation: navigationSlice
  },
});
