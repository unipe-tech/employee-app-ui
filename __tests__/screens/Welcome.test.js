import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { Provider } from "react-redux";
import WelcomePage from "../../screens/00_login/WelcomePage";
import { store } from "../../store/store";

let navigation;
beforeEach(() => {
  navigation = {
    dispatch: jest.fn(),
    navigate: jest.fn(),
  };
});

describe("Welcome Screen", () => {
  // snapshot test
  it("should render UI properly", () => {
    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <WelcomePage />
        </NavigationContainer>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // navigates to the next screen => Login Screen
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <Provider store={store}>
        <NavigationContainer>
          <WelcomePage navigation={{ navigate }} />
        </NavigationContainer>
      </Provider>
    );
    const event = fireEvent.press(getByText("Welcome!"));
    expect(navigate).toHaveBeenCalledWith("Login");
  });
});
