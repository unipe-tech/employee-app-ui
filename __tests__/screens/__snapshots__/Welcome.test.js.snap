// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen should render UI properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "process": [Function],
      }
    }
    style={
      Object {
        "height": 87,
        "marginLeft": 70,
        "marginTop": 30,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "Roboto",
        "fontSize": 18,
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 20,
      }
    }
  >
    Letâ€™s start onboarding process by verifying below documents.
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(1,0,0,0)",
          },
          Object {
            "flex": 1,
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(1,0,0,0)",
              "flexDirection": "row",
              "justifyContent": "space-around",
            },
            Object {
              "flexDirection": "column",
              "width": 30,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#4E46F1",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#aaaaaa",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#aaaaaa",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#aaaaaa",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#aaaaaa",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#aaaaaa",
                "borderRadius": 15,
                "borderWidth": 3,
                "height": 30,
                "justifyContent": "center",
                "overflow": "hidden",
                "width": 30,
                "zIndex": 2,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "justifyContent": "space-around",
            },
            Object {
              "flexDirection": "column",
              "paddingHorizontal": 4,
            },
            Object {
              "alignItems": "flex-start",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#4E46F1",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            Mobile Number
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#999999",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            Aadhaar Card
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#999999",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            PAN Card
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#999999",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            Bank Account
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#999999",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            Profile
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 12,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "color": "#999999",
                },
                Object {
                  "fontFamily": undefined,
                  "fontSize": 14,
                },
              ]
            }
          >
            Photo
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#6200EE",
        "borderBottomEndRadius": 4,
        "borderBottomStartRadius": 4,
        "borderTopEndRadius": 4,
        "borderTopStartRadius": 4,
        "fontSize": 20,
        "height": 50,
        "marginTop": 20,
        "padding": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 1.41,
        "width": 320,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomEndRadius": 4,
            "borderBottomStartRadius": 4,
            "borderTopEndRadius": 4,
            "borderTopStartRadius": 4,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onLayout={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "minWidth": 64,
              "paddingEnd": 16,
              "paddingStart": 16,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
              },
            ]
          }
        />
        <Text
          selectable={false}
          style={
            Array [
              Object {
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 1.25,
                "textTransform": "uppercase",
              },
              Object {
                "color": "#000000",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                },
                Object {
                  "opacity": 1,
                  "textTransform": "none",
                },
                undefined,
              ],
            ]
          }
        >
          Welcome!
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 30,
      }
    }
  />
</RCTSafeAreaView>
`;
